pyccel/parser/semantic.py

```diff
@@ -1095,7 +1097,7 @@ def _infere_type(self, expr, **settings):
             return self._infere_type(expr.value)
 
         elif isinstance(expr, IfTernaryOperator):
```

Comment:
Is an `IfTernaryOperator` the right-hand side expression of
```
a = 0 if b is None else 1
```
?

And if so, do we have unit tests for it?

Reply:
Yes exactly. There are some tests in tests/codegen/scripts/Functional_Stmts.py

Reply:
I know that this is not relevant to this pull request, but we should maybe check if this really works with pyccel/epyccel. If not we should create an issue.